### PR TITLE
options: nlopt_set_xtol_abs check xtol_abs

### LINKED ISSUE #461: null pointer dereference in nlopt_optimize_ / nlopt_set_xtol_abs
_nlopt_optimize_(...)_ contains a few _nlopt_set_xtol_abs(local_opt, opt->xtol_abs)_ calls with _opt->xtol_abs_ beeing null since #268. This causes a null pointer dereference in _nlopt_set_xtol_abs(...)'s_ memcpy.

### PR DESCRIPTION
**Pull request by jschueller**
Closes #461

### CODE REVIEW

**Comment by jschueller** [comment]
cc @stevengj

**Comment by jschueller** [comment]
ping @stevengj

File: src/api/options.c [hunk 1]
```diff
@@ -691,6 +691,12 @@ GETSET(ftol_rel, double, ftol_rel) GETSET(ftol_abs, double, ftol_abs) GETSET(xto
 {
     if (opt) {
         nlopt_unset_errmsg(opt);
+	if (!xtol_abs)
+	{
```
Comment (stevengj) [inline src/api/options.c:695]:
```suggestion
	if (!xtol_abs) {
```

File: src/api/options.c [hunk 2]
```diff
@@ -738,6 +743,12 @@ nlopt_result NLOPT_STDCALL nlopt_set_x_weights(nlopt_opt opt, const double *x_we
     if (opt) {
         unsigned i;
         nlopt_unset_errmsg(opt);
+	if (!x_weights)
+	{
```
Comment (stevengj) [inline src/api/options.c:747]:
```suggestion
	if (!x_weights) {
```

**Review by stevengj — APPROVED** [review]

**Pull request merged**